### PR TITLE
Haskell class post and layout adjustments for multiple authors

### PR DESCRIPTION
**Pull request by KamWis**
In order to correctly mark authors of the post some changes were required in `home`, `post` and `footer-single` htmls.

This PR also adds post for Haskell class.

### CODE REVIEW

File: _posts/2017-10-17-first-haskell-class-meeting.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+layout: post
+title:  "First Haskell class meeting"
+date: 2017-10-17 15:45:00 +0200
+categories: learn product
```
Comment (gilligan) [inline _posts/2017-10-17-first-haskell-class-meeting.md]: how about adding haskell here as tag?

File: _posts/2017-10-17-first-haskell-class-meeting.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+show_avatar : true
+read_time : 15
+show_related_posts: false
+square_related: recommend-kamil
+feature_image: posts/2017-10-17-first-haskell-class-meeting/haskell-poster.jpg
+---
+
+In HolidayCheck we believe in the value of learning new things. So the last week we held a first meeting of our Haskell Learning Group. The point of this group is to learn ourselves the basics of Haskell language.
+
+We do not intend to rewrite all our production codebase to Haskell (yet?) nor any particular parts of it. We just want to add yet another tool to our toolbelt to choose from. And Haskell is a great tool! It’s well established statically and strongly typed pure functional language, with a ton of research under it’s belt.
```
Comment (gilligan) [inline _posts/2017-10-17-first-haskell-class-meeting.md]: nicely put.

File: _posts/2017-10-17-first-haskell-class-meeting.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+But when you have a class you also need a book, and it should be an approachable book so everyone on any level of knowledge could easily learn from it. There are a lot of books to learn Haskell from, but we decided to go with [Haskell Programming from first principle][haskellbook].
+
+[haskellbook]: http://haskellbook.com/
+
+The book starts with the chapter on Lambda calculus, a mathematical concept that underlines functional programming, and as it’s not a requirement to know it to program in Haskell this after all `from the first principle`.
```
Comment (gilligan) [inline _posts/2017-10-17-first-haskell-class-meeting.md]: i find the sentence structure a bit confusing - could you maybe split it up in two sentences?

File: _posts/2017-10-17-first-haskell-class-meeting.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+Around 7 people were present at the class. One may suggest it was insignificant or  unimportant because of low attendance. It’s not true. As the matter of fact it was very fun and valuable!
+
+We started from small talk about what we liked in chapter 1. Everyone had different expressions about it. That allowed us to take other point of view on Lambda calculus then our own. Already a victory. True value, tough, came just after: resolving exercises.
```
Comment (gilligan) [inline _posts/2017-10-17-first-haskell-class-meeting.md]: `different expressions` - do you maybe mean `everyone had different thoughts about it` ?

File: _posts/2017-10-17-first-haskell-class-meeting.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+At first we went by the book. We took a look on `combinators`. That went very smoothly since we all knew more or less what `combinators` are. One of participants resolved related to `combinators` exercises. So, what are `combinators`? They are Lambdas without `free variables`. `Free variable` is the one which is not defined in Lambda’s head and it’s used in the Lambda’s body.
+
+Here we stopped for a second. We take our time to also point out the `Alpha equivalence`, which is when structure of given Lambda is exactly the same as some other Lambda, but variables are named differently. For instance:  𝜆𝑥.𝑥𝑥𝑥 and 𝜆.𝑦.𝑦𝑦𝑦
+
+After taking `Combinators` exercises like a champs we started talking about `normal and diverge Lambdas`. Here one of participants asked another to also show us on screen how he is deciding if given Lambda `diverges` or not. Person who was resolving this part of exercises did it very nicely and smoothly.
```
Comment (gilligan) [inline _posts/2017-10-17-first-haskell-class-meeting.md]: Maybe better `after mastering the Combinators exercises we started ..`

File: _posts/2017-10-17-first-haskell-class-meeting.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+reduction starts from far most left (if parentheses doesn’t apply otherwise),
+after each reduction we removing one head,
+reduction starting from outside to inside direction,
+inside reduction occurs only when there is nothing left to reduce outside,
+it’s possible to utilise `alpha equivalence` for convenience (ex. when we got two variables called `z` in `(𝜆z.z)(𝜆x.z)` we can call one of them `a` and we will get `(𝜆z.z)(𝜆x.a)`),
+When you stuck it’s good to write before each reduction what variable actually will become (ex. we reducing this: `(𝜆z.z)(𝜆x.z)`, we establish that `z =: (𝜆x.z)` so our result is `(𝜆x.z)`. There is nothing more to reduce.
+
+Beta reduction went nicely. Then we spoke about what we liked on this meeting and shared our thoughts. One of the conclusion was that Math can be very fun. Many people is afraid of Mathematics, but it appears it can give you pleasure and resolving Mathematical problems is very rewarding.
+
+So! Do not afraid of taking first steps. They may be hard, but reward is far greater than you expect!
```
Comment (gilligan) [inline _posts/2017-10-17-first-haskell-class-meeting.md]: -> So don't be afraid of taking those first steps

File: _posts/2017-10-17-first-haskell-class-meeting.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+To understand what `diverges` is one needs to know what `beta normal form` is, and for that I should probably tell you about `beta reduction`. So a quick recap just here, and if one would like a more proper explanation I suggest looking to a book mentioned earlier. `Beta reduction` is a process of applying function to its parameters, simple as that. `Beta normal form` is when there are no more parameters to apply. But sometime we get an infinite loop, where our expression never terminates no matter how much we `beta reduce` it, then we say it `diverges`.
+
+What was left from exercises was even more fun! We took `beta reduction` on board.
+Almost each of us made at least one exercise. We had precedent where one of attendants didn’t actually read the whole chapter. It didn’t stop us from explaining to him step by step `beta reduction` on those exercises. What even more cool was that we all valued from it and we built firm foundation of Lambda knowledge. We explained what Lambda head is, that it starts from `𝜆` and ends on `.`. We established that:
```
Comment (gilligan) [inline _posts/2017-10-17-first-haskell-class-meeting.md]: What even more cool was -> What was even cooler

File: _posts/2017-10-17-first-haskell-class-meeting.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
+---
+layout: post
+title:  "First Haskell class meeting"
```
Comment (HeeL) [inline _posts/2017-10-17-first-haskell-class-meeting.md]: `Haskell class` sounds like someone comes and teaches us. What we have is more like learning group, even though some people are far beyond from being beginners

**Comment by gilligan** [comment]
Hm.. well maybe switch it to “Haskell Learning Group” then. It’s the name i
used to introduce it with anyway.

**Comment by KamWis** [comment]
@gilligan @HeeL - are you alright with those changes? :)

**Comment by HeeL** [comment]
There are couple of places in the post and also on the image where it says "Haskell classes". It would be consistent to change it everywhere.

**Comment by KamWis** [comment]
@HeeL Is it fine for you now? :) Also @gilligan

**Comment by gilligan** [comment]
There is one last appearance of "class" which is the image with the logo that uses "classes"..  Maybe you could change that to "Haskell Learning Group" or whatever as well ?

**Comment by KamWis** [comment]
done :)

**Comment by gilligan** [comment]
LGTM - great work

**Comment by HeeL** [comment]
You forget to apply fixups with rebase interactive, so we have 2 fixup commits. Can we do anything about it or  will that require force push and rewriting history in  master?

**Pull request merged**